mmv1/products/firebaseappcheck/DeviceCheckConfig.yaml

```diff
@@ -0,0 +1,113 @@
+name: "DeviceCheckConfig"
+base_url: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig
+self_link: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig
+create_url: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig?updateMask=tokenTtl,keyId,privateKey
+create_verb: :PATCH
+update_verb: :PATCH
+update_mask: true
+skip_delete: true
+description: |
+  An app's DeviceCheck configuration object. Note that the Team ID registered with your
+  app is used as part of the validation process. Make sure your `google_firebase_apple_app` has a team_id present.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://firebase.google.com/docs/app-check"
+  api: "https://firebase.google.com/docs/reference/appcheck/rest/v1/projects.apps.deviceCheckConfig"
+import_format:
+  [
+    "projects/{{project}}/apps/{{app_id}}/deviceCheckConfig",
+    "{{project}}/{{app_id}}",
+    "{{app_id}}",
+  ]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebase_app_check_device_check_config_minimal"
+    # Need the time_sleep resource
+    pull_external: true
+    primary_resource_id: "default"
+    vars:
+      bundle_id: "bundle.id.devicecheck"
+      key_id: "Key ID"
+      private_key_path: "path/to/private-key.p8"
+      team_id: "9987654321"
+    test_vars_overrides:
+      # Don't add random suffix
+      private_key_path: '"test-fixtures/private-key.p8"'
+      team_id: '"9987654321"'
+    test_env_vars:
+      project_id: :PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebase_app_check_device_check_config_full"
+    # Need the time_sleep resource
+    pull_external: true
+    primary_resource_id: "default"
+    vars:
+      bundle_id: "bundle.id.devicecheck"
+      key_id: "Key ID"
+      private_key_path: "path/to/private-key.p8"
+      team_id: "9987654321"
+      token_ttl: "7200s"
+    test_vars_overrides:
+      # Don't add random suffix
+      private_key_path: '"test-fixtures/private-key-2.p8"'
+      team_id: '"9987654321"'
+      token_ttl: '"7200s"'
+    test_env_vars:
+      project_id: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: app_id
+    description: |
+      The ID of an
+      [Apple App](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.iosApps#IosApp.FIELDS.app_id).
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    description: |
+      The relative resource name of the DeviceCheck configuration object
+    output: true
+    pattern: projects/{{project}}/apps/{{app_id}}/deviceCheckConfig
+  - !ruby/object:Api::Type::String
+    name: tokenTtl
+    description: |
+      Specifies the duration for which App Check tokens exchanged from DeviceCheck artifacts will be valid.
+      If unset, a default value of 1 hour is assumed. Must be between 30 minutes and 7 days, inclusive.
```

Comment:
If this default is expected to remain the same long-term, `default_value` may provide a better UX - see https://googlecloudplatform.github.io/magic-modules/develop/permadiff/#default for details / caveats.

Reply:
Requesting for a valid key at the moment. I'll update the thread when ready.

Reply:
The default ttl value is expected to change. Since App Check is fairly new, 1 hour is more of an educated guess. As more information becomes available, we may change the default ttl to a more reasonable value.